soft_999_Jenish_Pate_fw_103

```diff
@@ -0,0 +1,73 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct CounterStorage {
+  uint16_t potentiometer_data;   // Stores data
+  AdcChannel address;            // Stores channel
+} CounterStorage;
+
+ void prv_button_interrupt_handler(const Gpiovoid *empty) {    
+      CounterStorage *storage = empty;
+      adc_read_converted(storage->potentiometer_data, &storage->address);
+      LOG_DEBUG("potentiometer data: %d\n", storage->potentiometer_data);
+      delay_ms(200);
+}
+
+int main(void) {
+      interrupt_init();       // Initializing Interrupts
+      gpio_init();            // Initializing GPIO
+      gpio__it_init():        // Initializing Interrupts 
+
+      GpioAddress button = { .port = GPIO_PORT_B, .pin = 2 };                     // Assigning Adress
+
+      GpioAddresss potentiometer_addr = { .port = GPIO_PORT_A, .pin = 6};          // Assigning Adress
+
+      GpioSettings pot_settings {                    // GPIO Settings 
+            GPIO_DIR_IN,
+            GPIO_STATE_LOW,
+            GPIO_RES_NONE,
+            GPIO_ALTFN_ANALOG,
+             };
+
+       GpioSettings s_button_settings {     // GPIO Settings Button Settings (The videos was very helpful)
```

Comment:
only use the `s_` prefix if it's actually a static value. Here it's just a variable in a function, so don't use `s_`.